### ISSUE TITLE
error: too many initializers for '__pthread_mutex_s' in priority_mutex.h on ARM

### ISSUE THREAD

**Issue by auscompgeek** [wpilibsuite/wpiutil#65]
This is… rather bizarre, but I'm getting this compile error when I try to build robotpy-cscore on a Raspberry Pi 3 running Arch Linux ARM (armv7). (Bizarre because I wouldn't expect glibc to break backwards compatibility badly.)

```
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -march=armv7-a -mfloat-abi=hard -mfpu=vfpv3-d16 -O2 -pipe -fstack-protector-strong -fno-plt -march=armv7-a -mfloat-abi=hard -mfpu=vfpv3-d16 -O2 -pipe -fstack-protector-strong -fno-plt -march=armv7-a -mfloat-abi=hard -mfpu=vfpv3-d16 -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/usr/include/python3.6m -I/home/alarm/.local/include/python3.6m -Icscore_src/src/main/native/include -Iwpiutil_src/src/main/native/include -I/usr/lib/python3.6/site-packages/numpy/core/include -I/usr/include/python3.6m -c cscore_src/src/main/native/cpp/cscore_cpp.cpp -o build/temp.linux-armv7l-3.6/cscore_src/src/main/native/cpp/cscore_cpp.o -DVERSION_INFO="2018.0.1" -s -g0 -std=c++14 -fvisibility=hidden
  cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
  In file included from wpiutil_src/src/main/native/include/support/mutex.h:12:0,
                   from cscore_src/src/main/native/cpp/UnlimitedHandleResource.h:17,
                   from cscore_src/src/main/native/cpp/Handle.h:17,
                   from cscore_src/src/main/native/cpp/cscore_cpp.cpp:19:
  wpiutil_src/src/main/native/include/support/priority_mutex.h:47:79: error: too many initializers for '__pthread_mutex_s'
         {0, 0, 0, 0, 0x20 | PTHREAD_MUTEX_RECURSIVE_NP, __PTHREAD_SPINS, {0, 0}}};
                                                                                 ^
  wpiutil_src/src/main/native/include/support/priority_mutex.h:76:73: error: too many initializers for '__pthread_mutex_s'
     pthread_mutex_t m_mutex = {{0, 0, 0, 0, 0x20, __PTHREAD_SPINS, {0, 0}}};
                                                                           ^
  error: command 'gcc' failed with exit status 1
```

**Comment by calcmogul** [comment]
```// Do the equivalent of setting PTHREAD_PRIO_INHERIT.
#ifdef __PTHREAD_MUTEX_HAVE_PREV
  pthread_mutex_t m_mutex = {{0, 0, 0, 0, 0x20, __PTHREAD_SPINS, {0, 0}}};
#else
  pthread_mutex_t m_mutex = {{0, 0, 0, 0x20, 0, {__PTHREAD_SPINS}}};
#endif
```
Maybe that version of pthread defines `__PTHREAD_MUTEX_HAVE_PREV` but uses the other struct layout?

**Comment by auscompgeek** [comment]
Here's a relevant section of /usr/include/pthread.h on this system:

```cpp
#if __PTHREAD_MUTEX_HAVE_PREV
# define PTHREAD_MUTEX_INITIALIZER \
  { { 0, 0, 0, 0, 0, __PTHREAD_SPINS, { 0, 0 } } }
# ifdef __USE_GNU
#  define PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP \
  { { 0, 0, 0, 0, PTHREAD_MUTEX_RECURSIVE_NP, __PTHREAD_SPINS, { 0, 0 } } }
#  define PTHREAD_ERRORCHECK_MUTEX_INITIALIZER_NP \
  { { 0, 0, 0, 0, PTHREAD_MUTEX_ERRORCHECK_NP, __PTHREAD_SPINS, { 0, 0 } } }
#  define PTHREAD_ADAPTIVE_MUTEX_INITIALIZER_NP \
  { { 0, 0, 0, 0, PTHREAD_MUTEX_ADAPTIVE_NP, __PTHREAD_SPINS, { 0, 0 } } }

# endif
#else
# define PTHREAD_MUTEX_INITIALIZER \
  { { 0, 0, 0, 0, 0, { __PTHREAD_SPINS } } }
# ifdef __USE_GNU
#  define PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP \
  { { 0, 0, 0, PTHREAD_MUTEX_RECURSIVE_NP, 0, { __PTHREAD_SPINS } } }
#  define PTHREAD_ERRORCHECK_MUTEX_INITIALIZER_NP \
  { { 0, 0, 0, PTHREAD_MUTEX_ERRORCHECK_NP, 0, { __PTHREAD_SPINS } } }
#  define PTHREAD_ADAPTIVE_MUTEX_INITIALIZER_NP \
  { { 0, 0, 0, PTHREAD_MUTEX_ADAPTIVE_NP, 0, { __PTHREAD_SPINS } } }

# endif
#endif
```

**Comment by auscompgeek** [comment]
I can also reproduce this on Fedora 27 (armv7). Fun times…

**Comment by auscompgeek** [comment]
Update: I cannot reproduce this on Debian buster for ARMv8. I'll investigate some more when I get the chance.

**Comment by PeterJohnson** [comment]
Should be fixed by wpilibsuite/allwpilib#1172.
